bzt/jmx/tools.py

```diff
@@ -678,8 +678,15 @@ def __gen_data_sources(self, scenario):
                 if not delimiter:
                     delimiter = guess_delimiter(source_path)
 
-            config = JMX._get_csv_config(source_path, delimiter, source.get("quoted", False), source.get("loop", True),
-                                         source.get("variable-names", ""))
+            random_parameter = source.get("random-order")
+            if not random_parameter:
+                config = JMX._get_csv_config(source_path, delimiter, source.get("quoted", False),
+                                             source.get("loop", True), source.get("variable-names", ""))
+            else:
+                config = JMX._get_csv_config_random(source_path, delimiter, source.get("loop", True),
+                                                    source.get("variable-names", ""), random_parameter,
+                                                    source.get("encoding", "UTF-8"),
+                                                    source.get("same-vals-for-all", True))
```

Comment:
It isn't good idea to have secret options ('encoding' and 'same-vals-for-all'). You should avoid it in code or add to documentation.